### PR TITLE
[otci] support ADB connection for OTCI

### PR DESCRIPTION
**Pull request by superwhd**

### CODE REVIEW

File: tools/otci/otci/command_handlers.py [hunk 1]
```diff
@@ -282,3 +282,48 @@ def wait(self, duration: float) -> List[str]:
 
     def set_line_read_callback(self, callback: Optional[Callable[[str], Any]]):
         self.__line_read_callback = callback
+
+
+class OtbrAdbCommandRunner(OTCommandHandler):
+
+    def __init__(self, host, port, username, password, sudo):
```
Comment (simonlingoogle) [inline tools/otci/otci/command_handlers.py:289]: sudo not used?

File: tools/otci/otci/command_handlers.py [hunk 2]
```diff
@@ -282,3 +282,48 @@ def wait(self, duration: float) -> List[str]:
...
+        from adb_shell.adb_device import AdbDeviceTcp
+
+        self.__host = host
+        self.__port = port
+        self.__adb = AdbDeviceTcp(host, port, default_transport_timeout_s=9.)
```
Comment (simonlingoogle) [inline tools/otci/otci/command_handlers.py:294]:
```suggestion
        self.__adb = AdbDeviceTcp(host, port, default_transport_timeout_s=9.0)
```
nit

File: tools/otci/otci/command_handlers.py [hunk 3]
```diff
@@ -282,3 +282,48 @@ def wait(self, duration: float) -> List[str]:
...
+
+        if cmd in ('reset', 'factoryreset'):
+            self.wait(3)
+
+        return output
+
+    def shell(self, cmd: str, timeout: float) -> List[str]:
+        output = self.__adb.shell(cmd, timeout_s=timeout)
+
+        return [l.strip('\r\n') for l in output.split('\r\n') if l]
```
Comment (simonlingoogle) [inline tools/otci/otci/command_handlers.py:319]: use splitlines ?

**Review by simonlingoogle — APPROVED** [review]
LGTM 👍 

Thanks

**Review by jwhui — APPROVED** [review]

**Pull request merged**